### PR TITLE
Feature ui update

### PR DESCRIPTION
**Pull request by KDuverge**
Made a few updates here

- Broke out the CSS to individual files
- Added more functionality in the JS file
- Cleaned up the HTML file and removed stuff that wasn't being used

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -9,27 +9,75 @@
...
+      </div>
+
+      <div class="now-playing">
+        <div class="now-playing-container">
           <label>Now Playing</label>
-          <div class="info" id="nowplayingInfo">
-            <div id="nowplayingImg"></div>
-            <div id="nowplayingTitle"></div>
+          <div class="now-playing-info">
+            <div class="now-playing-title">Blank Face</div>
```
Comment (dontfollowsean) [inline index.html]: Let's not hard code these values here

File: public/js/app.js [hunk 2]
```diff
@@ -1,45 +1,100 @@
...
-    lastPlayedDiv = document.getElementById("lastplayed"),
-    nowPlayingEndpoint = "https://35.211.223.26/nowplaying",
-    lastPlayedEndpoint = "https://35.211.223.26/lastplayed";
+const nowplayingImg = document.querySelector(".now-playing-img"),
+  nowplayingTitle = document.querySelector(".now-playing-title"),
+  nowplayingInfo = document.querySelector(".now-playing-info"),
+  nowPlayingArtist = document.querySelector(".now-playing-artist"),
+  recentlyPlayedDiv = document.querySelector(".recently-played-container"),
+  toggle = document.querySelector("#toggle"),
+  nowPlayingEndpoint = "https://35.211.223.26/nowplaying",
```
Comment (dontfollowsean) [inline public/js/app.js]: `http` for now. The server does not support https atm

Comment (KDuverge) [reply public/js/app.js]: alright cool, I'll change it back

File: index.html (hunk 1 above)
Comment (KDuverge) [reply index.html]: I have it hard coded for now, till Sarah designs what the UI will look like when there is no music playing

File: public/js/app.js [hunk 3]
```diff
@@ -1,45 +1,100 @@
...
+    "--theme-background-color":
+      "linear-gradient(0deg,rgba(0, 0, 0, 1) 0%, rgba(57, 57, 67, 1) 75%)",
+    "--theme-text-color": "#0085ff",
+    "--theme-single-hover-background-color": "#111112",
+    "--theme-single-background-color": "#18181d",
+    "--theme-single-shadow": "0px 2px 12px 2px rgba(0, 0, 0, 0.4)"
+  }
+};
+
+const toggleColor = e => {
```
Comment (dontfollowsean) [inline public/js/app.js]: nice

File: public/js/app.js [hunk 4]
```diff
@@ -1,45 +1,100 @@
...
+  nowplayingTitle = document.querySelector(".now-playing-title"),
+  nowplayingInfo = document.querySelector(".now-playing-info"),
+  nowPlayingArtist = document.querySelector(".now-playing-artist"),
+  recentlyPlayedDiv = document.querySelector(".recently-played-container"),
+  toggle = document.querySelector("#toggle"),
+  nowPlayingEndpoint = "https://35.211.223.26/nowplaying",
+  recentlyPlayedEndpoint = "https://35.211.223.26/recentlyplayed";
 
+const renderNowPlaying = song => {
+  const image = song.images.filter(i => i.width === 640);
```
Comment (dontfollowsean) [inline public/js/app.js]: Looking back at this, I think it's better to use `song.images.find(i => i.width === 640)` which will return a single image or null.

File: public/js/app.js [hunk 5]
```diff
@@ -1,45 +1,100 @@
...
+    "click",
+    () => (window.location.href = song.url)
+  );
+};
+
+const renderRecentlyPlayed = songs => {
+  let songInfoHtml = "";
+
+  songs.forEach(song => {
+    const image = song.images.filter(i => i.width === 64);
```
Comment (dontfollowsean) [inline public/js/app.js]: same here

Comment (KDuverge) [reply public/js/app.js]: copy that, I'll update now

File: public/js/app.js (hunk 4 above)
Comment (KDuverge) [reply public/js/app.js]: copy that, I'll update now

**Comment by KDuverge** [comment]
Had to update `image[0].url` to `image.url` as `.find` returns an object and not an array.

**Comment by KDuverge** [comment]
Updated Favicon, as well as updated the recently played font-size of the song title and artist

File: index.html (hunk 1 above)
Comment (dontfollowsean) [reply index.html]: ok we'll wait on that to merge then

**Comment by dontfollowsean** [comment]
> Had to update `image[0].url` to `image.url` as `.find` returns an object and not an array.

My bad I should have been clearer on that. Make sure you account for the null case as well, `.find` will return a single object or null. You'll throw an error if you try to read a property on an undefined object.

File: public/js/app.js [hunk 6]
```diff
@@ -1,45 +1,100 @@
...
+  nowPlayingArtist = document.querySelector(".now-playing-artist"),
+  recentlyPlayedDiv = document.querySelector(".recently-played-container"),
+  toggle = document.querySelector("#toggle"),
+  nowPlayingEndpoint = "http://35.211.223.26/nowplaying",
+  recentlyPlayedEndpoint = "http://35.211.223.26/recentlyplayed";
 
+const renderNowPlaying = song => {
+  const image = song.images.find(i => i.width === 640);
+
+  nowplayingImg.src = image[0].url;
```
Comment (dontfollowsean) [inline public/js/app.js]: Update this like you did in `renderRecentlyPlayed()`. Also check if null. If possible, find some generic cover art to show in cases where the image is null. Maybe something like [this](https://orig00.deviantart.net/26aa/f/2011/185/f/9/no_cover_itunes_by_stainless2-d3kxnbe.png)

Comment (KDuverge) [reply public/js/app.js]: copy that, I'll make those updates

Comment (KDuverge) [reply public/js/app.js]: if image returns null then ill render that generic cover art

File: public/js/app.js [hunk 7]
```diff
@@ -49,20 +52,29 @@ const renderRecentlyPlayed = songs => {
...
+    </div>
+    <div class="now-playing-img">
+      <img src="./public/images/generic-image.png" alt="Generic Cover Image" />
+    </div>
+  `;
 };
 
 const displaySongs = e => {
   fetch(nowPlayingEndpoint)
     .then(response => response.json())
```
Comment (dontfollowsean) [inline public/js/app.js]: Check here is the response has a 404 status, if it does display the no music message and album cover

File: public/js/app.js [hunk 8]
```diff
@@ -49,20 +52,29 @@ const renderRecentlyPlayed = songs => {
   TODO - 
     Proper error handling
 */
-const handleError = err => {
-  console.log(err);
+const handleNowPlayingError = err => {
```
[the remark this answers is not in the document]
Comment (dontfollowsean) [reply public/js/app.js]: For now you can log the error. Maybe in the future we can display a toast or something.

File: public/js/app.js (hunk 7 above)
Comment (KDuverge) [reply public/js/app.js]: If the response is 404, wouldn't that enter that catch block? So I can check to see if the status is 404 in the catch block and render the no music message?

File: public/js/app.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (KDuverge) [reply public/js/app.js]: copy that

**Pull request merged**